recipes/libyaml/all/conanfile.py

```diff
@@ -0,0 +1,68 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+class LibYAMLConan(ConanFile):
+    name = "libyaml"
+    description = "LibYAML is a YAML parser and emitter library."
+    topics = ("conan", "libyaml", "yaml", "parser", "emitter")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/yaml/libyaml"
+    license = "MIT"
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {"shared": [True, False], "fPIC": [True, False]}
+    default_options = {"shared": False, "fPIC": True}
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _build_subfolder(self):
+        return "build_subfolder"
+
+    def config_options(self):
+        if self.settings.os == 'Windows':
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
+        del self.settings.compiler.cppstd
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version])
+        extracted_dir = '{}-{}'.format(self.name, self.version)
+        os.rename(extracted_dir, self._source_subfolder)
+
+    def _configure_cmake(self):
+        cmake = CMake(self)
+        cmake.definitions["BUILD_TESTING"] = False
+        cmake.definitions["INSTALL_CMAKE_DIR"] = 'lib/cmake/libyaml'
+        cmake.definitions["YAML_STATIC_LIB_NAME"] = "yaml"
+        cmake.configure(build_folder=self._build_subfolder)
+        return cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package(self):
+        # 0.2.2 has LICENSE, 0.2.5 has License, so ignore case
+        self.copy(pattern="License", dst="licenses",
+                  src=self._source_subfolder, ignore_case=True)
+        cmake = self._configure_cmake()
+        cmake.install()
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "libyaml",
+                               "yamlConfig.cmake"))
+        os.unlink(os.path.join(self.package_folder, "lib", "cmake", "libyaml",
+                               "yamlConfigVersion.cmake"))
```

Comment:
so that leaves in the package folder:

```
lib/cmake/libyaml/yamlTargets-release.cmake
lib/cmake/libyaml/yamlTargets.cmake
```

Reply:
Just do:
`tools.rmdir(os.path.join(self.package_folde, "lib", "cmake")` in `package`

It looks like the library is exported as `yaml`. [See this reference](https://github.com/yaml/libyaml/blob/acd6f6f014c25e46363e718381e0b35205df2d83/CMakeLists.txt#L91)
So you'll probably want to use components.
The code below should provide this (untested):
```python
def package_info(self):
    self.cpp_info.components["yaml"].libs = ["yaml"]
    if self.settings.compiler == "Visual Studio":
         self.cpp_info.components["yaml"].defines = ["YAML_DECLARE_EXPORT" if self.options.shared else "YAML_DECLARE_STATIC"]
    self.cpp_info.names["cmake_find_package"] = "yaml"
    self.cpp_info.names["cmake_find_package_multi"] = "yaml"
```
That way:
- no need for `tools.collect_libs` (=sign of lazy programming)
- no need for builddirs (the cmake scripts might contain paths of the build machine)
- the hooks will be happy (and if the hooks are happy, we are happy)

Reply:
Done.